### PR TITLE
Issue #37 Use Hardlinks instead of copy

### PR DESCRIPTION
**Pull request by possi**
If destination is on same filesystem (according to stat()) as the source
(our vendor-dir), we can create a hardlink for faster copy and less disk
usage.

Optional Todo: Create an config setting to read from composer to enable or disable hardlinking to allow users to fix problems under certain circumstances

### CODE REVIEW

**Comment by possi** [comment]
Related to PR #37

**Comment by AydinHassan** [comment]
Hi @possi, sorry it took a while to get here, I've been travelling a fair bit recently. I don't mind this change but I'd like to ask a couple of things. Can we:

1. Put it behind a configuration option
2. Disable by default
3. Add some test coverage

I ask of these as this is the first request of the sort in over 2 years, it seems like it's a edge-case and technically it is BC break - albeit a very small one. Putting it behind a config option would allow to bring it in without bumping the major version.

File: src/CoreInstaller.php [hunk 1]
```diff
@@ -76,12 +76,34 @@ public function install($source, $destination)
...
     }
 
     /**
+     * @param string $item
+     * @param string $destinationFile
+     */
+    protected function installFile($item, $destinationFile)
+    {
+        // Hardlinks require to be on same Filesystem/Blockdevice
+        list($source_dev) = stat($item);
```
Comment (AydinHassan) [inline src/CoreInstaller.php]: I think it would more performant and more readable if the check for same filesystem was performed only once at the start of the install process. If not on the same FS we can print a debug message stating that we are falling back to symlink.

**Comment by AydinHassan** [comment]
closing due to no updates, feel free to re-open if you want to continue working on this at some point :)
